test/partitioning.jl

```diff
@@ -196,7 +196,31 @@
   end
 
   @testset "SpatialPredicatePartitioner" begin
-    # TODO
+    g = RegularGrid{Float64}(10,10)
+    R = 5.0
+    SPred1(x,y) = norm(x-y) < R
+
+    p = partition(g, SpatialPredicatePartitioner(SPred1))
+    s = subsets(p)
+    testArray = sample(s)
```

Comment:
This test should be exhaustive for all sets in `s`. Please write a for loop instead of sampling a single set.